spec/api_client.js

```diff
@@ -75,45 +75,45 @@ describe('api client', function () {
     apiClient = new ApiClient(urlBase, serviceId, apiKeyId);
     apiClient.post(path, data)
       .then(function (response) {
-        expect(response.statusCode).to.equal(200);
+        expect(response.status).to.equal(200);
         done();
     });
   });
 
   it('should direct get requests through the proxy when set', function (done) {
     var urlBase = 'https://api.notifications.service.gov.uk',
-      proxyUrl = 'http://proxy.service.gov.uk:3030/',
+      proxyConfig = { host: 'addressofmyproxy.test'},
       path = '/email',
       apiClient = new ApiClient(urlBase, 'apiKey');
 
-    nock(urlBase)
-      .get(path)
+    nock("http://" + proxyConfig.host)
```

Comment:
This change seems to be from mocking calls to the API to mocking calls to the proxy. I may have misunderstood as this is a bit out of my experience but is this intended?

Reply:
Yep, this is intended.

So something about how axios does things compared to requests means that it wasn't working when mocking the endpoint. It was actually making a real http request to the mock address which would fail on the DNS lookup and cause the test to error.

What happens when you make a request via a proxy is we actually send a single http request to

`http://addressofmyproxy.test/https://api.notifications.service.gov.uk/email`

which would then have the proxy make a request to `https://api.notifications.service.gov.uk/email` and return that response as it's response to the original request.

Therefore, as we know this we can instead mock `addressofmyproxy.test` to avoid any real http requests being made, whilst still returning 200.

Reply:
Makes sense.